system/src/system_update.cpp

```diff
@@ -93,6 +93,46 @@ const char* flag_to_string(uint8_t flag) {
     return flag ? "true" : "false";
 }
 
+using particle::protocol::message_handle_t;
+
+const message_handle_t MESSAGE_HANDLE_INVALID = -1;
```

Comment:
Can we guarantee that this ID will never be assigned to a valid message (e.g. by using a larger signed type for `message_handle_t`), or reimplement the client code so that it doesn't depend on magic values?

Reply:
Yes totes, this needs fixing, but it's not new - -1 was used rather than a constant. 

I'll add checks in the message_id generator that it excludes 0 and 65535, and message ids will start close to the maximum so we can see the effects of rollover.

Reply:
> but it's not new - -1 was used rather than a constant

Could you elaborate on where it was used?

Reply:
I've removed the use of the constant in the latest commits.